src/main/scala/mesosphere/marathon/cron/CronMonitorActor.scala

```diff
@@ -0,0 +1,94 @@
+package mesosphere.marathon.cron
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Cancellable, Props}
+import mesosphere.marathon.SchedulerActions
+import mesosphere.marathon.core.launchqueue.LaunchQueue
+import mesosphere.marathon.state.AppDefinition
+import mesosphere.marathon.state.schedule.Periodic
+
+import scala.concurrent.duration._
+
+
+class CronMonitorActor private(
+                                createSchedulerActions: ActorRef => SchedulerActions,
+                                launchQueue: LaunchQueue,
+                                period: FiniteDuration
+                              ) extends Actor with ActorLogging {
+  import CronMonitorActor._
+  import Messages._
+  import Messages.Private._
+
+  // Actor State
+  private var monitorTick: Option[Cancellable] = None
+  private var schedulerActions: SchedulerActions = _
+  // End of actor state
+
+  //TODO: Leverage instance tracker to avoid launching already running tasks
+  //TODO: Manage schedule expression so no all tasks will be launched after the period
+
+  override def preStart(): Unit = {
+    monitorTick = Some(context.system.scheduler.schedule(period, period, self, PerformCheck))
+    schedulerActions = createSchedulerActions(self)
+  }
+
+  private def receiveCommands: Receive = {
+    case PerformCheck =>
+      launchCronApps
+  }
+
+  private def receiveInternalEvents: Receive = {
+    case LaunchPeriodicTasks(taks) =>
+      taks foreach { runSpec =>
+        launchQueue.add(runSpec)
```

Comment:
I've just noticed this call blocks in when using `LaunchQueueDelegate` implementation and, therefore it should be performed within a Future. This could be done at `launchCronApps`, that way, the future for the result of the request to the group repository could be composed with it via `flatMap`